### ISSUE TITLE
[Enhancement] Mark extension source of extended enumerants

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#444]
Discussing #442 gave a small enhancement idea:
It would be nice to have source extension shown directly with extended enumerants. That meta-information could be seen in `vulkan.h` as well as in the specification text (which seem to list enums directly as C listing)

The change should not break anything. And on the face of it should be a relatively small change in the generator script(s).

For example, `VkResult` could look like this:
```
typedef enum VkResult {
    VK_SUCCESS = 0,
    VK_NOT_READY = 1,
    VK_TIMEOUT = 2,
    VK_EVENT_SET = 3,
    VK_EVENT_RESET = 4,
    VK_INCOMPLETE = 5,
    VK_ERROR_OUT_OF_HOST_MEMORY = -1,
    VK_ERROR_OUT_OF_DEVICE_MEMORY = -2,
    VK_ERROR_INITIALIZATION_FAILED = -3,
    VK_ERROR_DEVICE_LOST = -4,
    VK_ERROR_MEMORY_MAP_FAILED = -5,
    VK_ERROR_LAYER_NOT_PRESENT = -6,
    VK_ERROR_EXTENSION_NOT_PRESENT = -7,
    VK_ERROR_FEATURE_NOT_PRESENT = -8,
    VK_ERROR_INCOMPATIBLE_DRIVER = -9,
    VK_ERROR_TOO_MANY_OBJECTS = -10,
    VK_ERROR_FORMAT_NOT_SUPPORTED = -11,
    VK_ERROR_FRAGMENTED_POOL = -12,
    /* VK_KHR_surface */
    VK_ERROR_SURFACE_LOST_KHR = -1000000000,
    VK_ERROR_NATIVE_WINDOW_IN_USE_KHR = -1000000001,
    /* VK_KHR_swapchain */
    VK_SUBOPTIMAL_KHR = 1000001003,
    VK_ERROR_OUT_OF_DATE_KHR = -1000001004,
    /* VK_KHR_display_swapchain */
    VK_ERROR_INCOMPATIBLE_DISPLAY_KHR = -1000003001,
    /* VK_EXT_debug_report */
    VK_ERROR_VALIDATION_FAILED_EXT = -1000011001,
    /* VK_NV_glsl_shader */
    VK_ERROR_INVALID_SHADER_NV = -1000012000,
    /* VK_KHR_maintenance1 */
    VK_ERROR_OUT_OF_POOL_MEMORY_KHR = -1000069000,
} VkResult;
```

**Comment by oddhack** [comment]
There isn't interest in this inside Khronos at present; we think the fact that it's obvious when something is an extension enumerant + needing to look it up in the spec anyway to know what it does, should be sufficient. But it's certainly fairly straightforward to do in the generator scripts if you want to make such a header for yourself.

**Comment by NicolBolas** [comment]
It should also be noted that extension enumerators have the source extension number built into the number itself. So even with just the number, it seems clear which enums belong to the same extension. Obviously, you don't know a-priori which extension is associated with which number. But the number itself gets most of the info out there.

**Comment by krOoze** [comment]
@oddhack Something like this was done for structs and functions now in 1.2.142.
Given that, should this be revisited.

**Comment by oddhack** [comment]
@krooze, yes, it's on the list to be done fairly soon.

**Comment by krOoze** [comment]
This was implemented in 1.2.146 update.
